sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/DnsResolverTests.cs

```diff
@@ -0,0 +1,643 @@
+                parameters: dnsResolver);
+
+            createdDnsResolver.Should().NotBeNull();
+            createdDnsResolver.Id.Should().Be(TestDataGenerator.GenerateDnsResolverArmId(subscriptionId: this.SubscriptionId, resourceGroupName: resourceGroupName, dnsResolverName: dnsResolverName));
+            createdDnsResolver.Name.Should().Be(dnsResolverName);
+            createdDnsResolver.Tags.Should().BeNull();
+            createdDnsResolver.Location.Should().Be(Constants.DnsResolverLocation);
+            createdDnsResolver.Etag.Should().NotBeNullOrEmpty();
+            createdDnsResolver.VirtualNetwork.Id.Should().Be(dnsResolver.VirtualNetwork.Id);
+            createdDnsResolver.NumberOfInboundEndpoints.Should().Be(0);
+            createdDnsResolver.ProvisioningState.Should().Be(Constants.ProvisioningStateSucceeded);
+        }
+
+        [Fact]
+        public void PutDnsResolver_ResolverNotExistsWithTags_ExpectResolverCreated()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+            var dnsResolverTags = TestDataGenerator.GenerateTags();
+            var dnsResolver = TestDataGenerator.GenerateDnsResolver(location: Constants.DnsResolverLocation, subscriptionId: this.SubscriptionId, resourceGroupName: resourceGroupName, tags: dnsResolverTags);
+
+            var createdDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: dnsResolverName,
+                parameters: dnsResolver);
+
+            createdDnsResolver.Should().NotBeNull();
+            createdDnsResolver.Tags.Should().NotBeNull().And.BeEquivalentTo(dnsResolverTags);
+        }
+
+        [Fact]
+        public void PutDnsResolver_ResolverNotExistsWithMalformedVirtualNetworkArmId_ExpectError()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+            var dnsResolver = TestDataGenerator.GenerateDnsResolver(location: Constants.DnsResolverLocation, subscriptionId: this.SubscriptionId, resourceGroupName: resourceGroupName);
+            dnsResolver.VirtualNetwork.Id = TestDataGenerator.GetRandomString(15);
+
+            Action createdDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: dnsResolverName,
+                parameters: dnsResolver);
+
+            createdDnsResolverAction.Should().NotBeNull();
+            createdDnsResolverAction.Should().Throw<CloudException>().Which.Response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        }
+
+
+        [Fact]
+        public void PutDnsResolver_VirtualNetworkAttachedToAnotherVirtualNetwork_ResolvereNotExists_ExpectConflict()
```

Comment:
These are very long method names - this could lead to issues with the path length for the recording files (which track to the class and method name of the corresponding tests)

Reply:
Updated to `PutDnsResolver_VirtualNetworkIsUsedElsewhere_ExpectConflict`